hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/BaseTenants.java

```diff
@@ -46,7 +46,7 @@ private BaseTenants() {
 
     }
 
-    public static class ReadWrite<BE> extends Mutator<BE, Tenant, Tenant.Blueprint, Tenant.Update>
+    public static class ReadWrite<BE> extends Mutator<BE, Tenant, Tenant.Blueprint, Tenant.Update, String>
```

Comment:
not sure I like the string "String" as a type parameter, you basically overrides the basic java type with it, what if I wanted to add simple helper method which will use java.lang.String? I'd have to use the fully qualified path, this may be confusing, ID was imho better.